### ISSUE TITLE
ToolShed workflow SVG display broken

### ISSUE THREAD

**Issue by peterjc** [galaxyproject/galaxy#623]
Steps to reproduce:
1. Go to the Tool Shed https://toolshed.g2.bx.psu.edu or Test Tool Shed https://testtoolshed.g2.bx.psu.edu
2. Click "Search for workflows" (top of left pane, under "Search")
3. Click "Search repositories" (no search text so get all of them)
4. Pick a repository, e.g. blast_top_hit_species
5. Under "Contents of this repository", click on the workflow
6. View mockup of the workflow.

Or more directly,
1. Click on https://toolshed.g2.bx.psu.edu/view/peterjc/blast_top_hit_species
2. Under "Contents of this repository", click on the "Species of top BLAST hits" workflow
3. View mockup of the workflow.

The resulting page has this text:

```
Species of top BLAST hits
Boxes are red when tools are not available in this repository
(this page displays SVG graphics)
```

_Expected results:_ Inline SVG graphic

_Safari results:_ No image, no error

_Firefox results:_ No image, error text:

```
XML Parsing Error: no element found
Location: https://toolshed.g2.bx.psu.edu/repository/generate_workflow_image?workflow_name=77e15ba155420f0527c13a1d005a5e6360ba4f06%3A53706563696573206f6620746f7020424c4153542068697473&repository_metadata_id=7f60e3e1f5201579
Line Number 1, Column 1:
^
```

Spotted by @jj-umn

**Comment by dannon** [comment]
Thanks for the detailed report, I'm going to take a look now.

**Comment by martenson** [comment]
We have been looking at this yesterday with @davebx, not solved yet though. 

At `dev` it works localy but TTS is on `dev` and does not work.

**Comment by peterjc** [comment]
Thanks - I suspect this is a recent regression but don't think I can help narrow down when this broke.

**Comment by dannon** [comment]
There's technically another spec-breaking error in the XML we generate that I'm also working on, but the fix in #624 does get them rendering again for me.

**Comment by dannon** [comment]
#624 has been updated with a second commit fixing the other issue I mentioned and this should be resolved when that is tested/merged.

**Comment by dannon** [comment]
This is resolved with #624

**Comment by martenson** [comment]
thanks @dannon
